soc/arm/nordic_nrf/nrf53/Kconfig.defconfig.nrf5340_CPUAPP_QKAA

```diff
@@ -11,4 +11,17 @@ config SOC
 config NUM_IRQS
 	default 69
 
+config NRF5340_CPUAPP_ERRATUM19
+	bool "Include workarounds for Erratum 19"
+	default y
```

Comment:
This shall not be defined for non-secure builds.
Maybe add a dependence on !TRUSTED_EXECUTION_NONSECURE?

Reply:
What if the application needs to know the granularity of the locking mechanism? Say that an application needs to check if an operation is within bounds or not of the locking granularity of its onboard SPU, then this information would have to be present in the non-secure code as well. 

Also, I don't see any negative effect of including this in the non secure build, but I might be wrong :)

Reply:
This is a very hypothetical problem, with no applicable use-case, so far, to the best of my knowledge. In theory a Non-Secure Zephyr build should operate with no reference to the SPU configuration; SPU is to be used and referred to only from secure-type firmware(s).

There is one case, where the SPU granularity affects the Non-Secure build; that's the MPU user-accessibility evaluation, done in `mpu_buffer_validate()`. and that one is already taking into account the existence of SPU.

The negative impact is a code size and boot delay increase due to executing the added SYS_INIT function. That's minor, of course, but I also think Kconfig should not bring in definitions we don't need when building a zephyr image, for better controlling the build.
However, if we come at a point that this information is needed by non-secure builds, we can easily add it by removing the dependency. I just don't want to do that in advance.

Reply:
For technical reasons I think we should not have a `depends on !TRUSTED_EXECUTION_NONSECURE`, since Partition Manager is run based on the app's Kconfig.

Reply:
Well, the upstream project must be independent of any downstream constraints.

Reply:
If you need more of an argument: SPU does non-trustzone things (RWX control) which it's theoretically possible the non-secure app must know about.

Reply:
@oyvindronningstad @hakonfam 

Exposing SPU configuration awareness to Non-Secure, sort of defeats the principle of Non-Secure firmware being non-aware to security configuration: such firmware should be _free_ to access resources it has been allocated, and _prevent_ itself from using resources it has not been explicitly allocated. That's why I want to block running Non-Secure firmware to know things around the SPU.

> If you need more of an argument: SPU does non-trustzone things (RWX control) which it's theoretically possible the non-secure app must know about.

It does that, I agree. But the Non-Secure firmware can always evaluate the applied memory access (Read/Write/eXecute/Secure) policies at runtime using the TT intrinsics; that should be the way to handle these challenges, IMHO

Reply:
@oyvindronningstad @hakonfam 

Exposing SPU configuration awareness to Non-Secure, sort of defeats the principle of Non-Secure firmware being non-aware to security configuration: such firmware should be _free_ to access resources it has been allocated, and _prevent_ itself from using resources it has not been explicitly allocated. That's why I want to block running Non-Secure firmware to know things around the SPU.

> If you need more of an argument: SPU does non-trustzone things (RWX control) which it's theoretically possible the non-secure app must know about.

It does that, I agree. But the Non-Secure firmware can always evaluate the applied memory access (Read/Write/eXecute/Secure) policies at runtime using the TT intrinsics; that should be the way to handle these challenges, IMHO (Let me know if I am mistaken on this)

Reply:
> such firmware should be _free_ to access resources it has been allocated

To do that it needs to know which resources it has been allocated, i.e. which peripherals, and also which memory areas.

> can always evaluate the applied memory access (Read/Write/eXecute/Secure) policies at runtime using the TT intrinsics;

I don't think the TT instruction can be used for this. If it can, you have to teach me :D.

Reply:
Everything that the Non-Secure image needs to know at build time should be passed as Kconfig options, that's agreed. But why does the Non-Secure image need to know an internal property of a peripheral that it cannot access? If the SPU-granularity affects other stuff (stupid example: FLASH_IMAGE_START_ADDRESS), then _those_ other stuff shall be passed on to the Non-Secure build directly.

The Non-Secure image may need, at run-time, an evaluation of memory addresses and address-ranges, so it protects-itself from causing secure faults. This is done via the TT intrinsics, but: TTs can also assess RWX as far as I can see:

```
typedef union {
	struct cmse_address_info {
		unsigned mpu_region:8;
		unsigned sau_region:8;
		unsigned mpu_region_valid:1;
		unsigned sau_region_valid:1;
		unsigned read_ok:1;
		unsigned readwrite_ok:1;
		unsigned nonsecure_read_ok:1;
		unsigned nonsecure_readwrite_ok:1;
		unsigned secure:1;
		unsigned idau_region_valid:1;
		unsigned idau_region:8;
	} flags;
	unsigned value;
} cmse_address_info_t;
```
That is, R/W/X/S policy can be evaluated by NS firmware at run-time without awareness of the IDAU properties. This alleviates the need for a dependency between Secure and Non-Secure FW images, so they can be developed independent from each other. We do not want to violate this ARM convention for NRF platforms in the upstream project.